### PR TITLE
Save search counts

### PR DESCRIPTION
**Pull request by crufter**

### CODE REVIEW

File: publicapi/handler/explore.go [hunk 1]
```diff
@@ -159,8 +172,38 @@ func (e *Explore) exploreList() ([]*API, error) {
 	return e.loadCache()
 }
 
+func (e *Explore) recordSearch(searchTerm string) error {
+	searchTerm = strings.ToLower(searchTerm)
+	searchTerm = strings.Replace(searchTerm, " ", "-", -1)
+
+	if searchTerm == "" {
+		return fmt.Errorf("no search term to track")
```
Comment (asim) [inline publicapi/handler/explore.go:180]: If an error is returned every call to /explore will log an error

File: publicapi/handler/explore.go [hunk 2]
```diff
@@ -159,8 +172,38 @@ func (e *Explore) exploreList() ([]*API, error) {
...
+		return err
+	}
+	if len(oldTrack) == 0 {
+		return e.trackSearch.Create(SearchCount{
+			Id:    searchTerm,
+			Count: 1,
+		})
+	}
+	tr := oldTrack[0]
+	logger.Infof("Found %v %v", tr.Id, tr.Count)
```
Comment (asim) [inline publicapi/handler/explore.go:194]: More useful logging message? so we don't just read Found x x in the log

File: publicapi/handler/explore.go [hunk 3]
```diff
@@ -159,8 +172,38 @@ func (e *Explore) exploreList() ([]*API, error) {
...
+func (e *Explore) recordSearch(searchTerm string) error {
+	searchTerm = strings.ToLower(searchTerm)
+	searchTerm = strings.Replace(searchTerm, " ", "-", -1)
+
+	if searchTerm == "" {
+		return fmt.Errorf("no search term to track")
+	}
+	oldTrack := []*SearchCount{}
+	err := e.trackSearch.Read(model.QueryEquals("id", searchTerm), &oldTrack)
+	if err != nil {
```
Comment (asim) [inline publicapi/handler/explore.go:184]: Will this just not return "not found"?

Comment (crufter) [reply publicapi/handler/explore.go:184]: No, it will return an empty slice

**Pull request merged**